src/sentry/workflow_engine/endpoints/validators/metric_alert_detector.py

```diff
@@ -0,0 +1,31 @@
+from rest_framework import serializers
+
+from sentry.snuba.snuba_query_validator import SnubaQueryValidator
+from sentry.workflow_engine.endpoints.validators.base import (
+    BaseGroupTypeDetectorValidator,
+    NumericComparisonConditionValidator,
+)
+from sentry.workflow_engine.models.data_condition import Condition
+from sentry.workflow_engine.types import DetectorPriorityLevel
+
+
+class MetricAlertComparisonConditionValidator(NumericComparisonConditionValidator):
+    supported_conditions = frozenset((Condition.GREATER, Condition.LESS))
+    supported_results = frozenset((DetectorPriorityLevel.HIGH, DetectorPriorityLevel.MEDIUM))
+
+
+class MetricAlertsDetectorValidator(BaseGroupTypeDetectorValidator):
+    data_source = SnubaQueryValidator(required=True)
+    data_conditions = MetricAlertComparisonConditionValidator(many=True)
+
+    def validate(self, attrs):
```

Comment:
i _really_ like the composition of this validator now -- i feel like this will be way more reusable of a class as we introduce more detectors! 🎉